storagemarket/impl/provider.go

```diff
@@ -146,6 +153,29 @@ func NewProvider(net network.StorageMarketNetwork,
 		return nil, err
 	}
 
+	indexer.RegisterCallback(func(ctx context.Context, contextID []byte) (provider.MultihashIterator, error) {
```

Comment:
I'm wondering if it's safe to register the callback before the dagstore has started 🤔